Automattic/woocommerce-payments
# Adds a new option which reflects the modal being dismissed.

Fixes #7683, #7544

#### Changes proposed in this Pull Request



* Adds a new option to reflect the eligibility modal being dismissed.
* Update the logic to take this new option into account.
* Update the logic to clear this option when a new account onboarding starts.







#### Testing instructions





* Delete any previously created account associated with your store. (or update the `wcpay_accounts` table on `localhost:8087` to set `account_type` to something other than `wcpay` to make the system not recognise it and then clear the cache.
* Run the server listener with `npm run listen`
* Initiate a new onboarding flow - select country = `US`, business_type = `individual`, store readiness to `ready now` and TPV to the lowest amount to trigger Progressive Onboarding.
* Complete the PO flow to onboard a PO account.
* Verify that you are redirected to the Overview page, and that the PO eligibility modal displays.

#### Closing the modal:

* Close the modal using the `X` icon at the top right.
* In another tab, go to `localhost:8083` and check the `wp_options` table, looking for the `option_name` = `wcpay_onboarding_eligibility_modal_dismissed` option. It should be set to `1`.
* Refresh the page, and the eligibility modal should no longer show.
* Delete the option, then refresh the page again. The modal should now show again.

#### Selecting Payments Only

* Click the `Enable payments only` option. The modal should close.
* In another tab, go to `localhost:8083` and check the `wp_options` table, looking for the `option_name` = `wcpay_onboarding_eligibility_modal_dismissed` option. It should be set to `1`.
* Refresh the page, and the eligibility modal should no longer show.
* Delete the option, then refresh the page again. The modal should now show again.

#### Enabling Deposits

* Click the `Set up payments and deposits` option. You should be redirected to the Stripe KYC.
* In another tab, go to `localhost:8083` and check the `wp_options` table, looking for the `option_name` = `wcpay_onboarding_eligibility_modal_dismissed` option. It should be set to `1`.
* Complete the onboarding to enable deposits
* You should be redirected back to the Overview page, and the eligibility modal should not display.


 

-------------------


**Post merge**
